src/screenComponents/viewport3d.cpp

```diff
@@ -547,37 +543,21 @@ void GuiViewport3D::onDraw(sf::RenderTarget& window)
 sf::Vector3f GuiViewport3D::worldToScreen(sf::RenderTarget& window, sf::Vector3f world)
 {
     world -= camera_position;
+    auto view_pos = model_matrix * glm::vec4{ world.x, world.y, world.z, 1.f };
+    auto pos = projection_matrix * view_pos;
```

Comment:
I understand this bit may feel "extra" but please bear with me here - the view matrix is next in line to completely switch over.
This little intermediate step avoids me a few hoop while keeping the PRs "bite-sized" (I'm currently backporting the work from my sdl branch, and splitting it up so the PRs remain reasonable).